### PR TITLE
Allow client to enqueue tasks with a unique option

### LINKED ISSUE #119: [FEATURE REQUEST] Support unique jobs
**Is your feature request related to a problem? Please describe.**
Any chance we would be supporting this? More details can be found [here](https://github.com/mperham/sidekiq/wiki/Ent-Unique-Jobs). Thanks.

### PR DESCRIPTION
**Pull request by hibiken**
Closes #119 
Design Doc: https://docs.google.com/document/d/17PHzGqsEEuhhrmyZLzVSR24Z7g5YJPs1n38Ec2-63WU/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ git:
   depth: 1
 env:
   - GO111MODULE=on # go modules are the default
```
Comment (cayter) [inline .travis.yml]: We can remove this since it's enabled by default starting 1.13?

Comment (hibiken) [reply .travis.yml]: Done. Thank you for pointing this out!

File: client.go [hunk 2]
```diff
@@ -70,11 +74,29 @@ func Deadline(t time.Time) Option {
 	return deadlineOption(t)
 }
 
+// Unique returns an option to acquire a uniqueness lock for the given task.
+// The lock will be released when the task gets processed successfully or TTL has expired.
+//
+// Uniqueness of a task is based on the following properties:
+//     - Task type
+//     - Task payload
+//     - Queue name
```
Comment (hibiken) [inline client.go]: Improve this documentation

File: client.go [hunk 3]
```diff
@@ -70,11 +74,29 @@ func Deadline(t time.Time) Option {
...
+func Unique(ttl time.Duration) Option {
+	return uniqueOption(ttl)
+}
+
+// ErrDuplicateTask indicates that the given task could not be enqueued since it's a duplicate
+// of another task.
+//
+// ErrDuplicateTask error only applies to tasks enqueued with a Unique option.
```
Comment (hibiken) [inline client.go]: Improve this documentation

File: client_test.go [hunk 4]
```diff
@@ -361,3 +363,210 @@ func TestClientEnqueueIn(t *testing.T) {
...
+		h.FlushDB(t, r) // clean up db before each test case.
+
+		// Enqueue the task first. It should succeed.
+		err := c.Enqueue(tc.task, Unique(tc.ttl))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		gotTTL := r.TTL(uniqueKey(tc.task, tc.ttl, base.DefaultQueueName)).Val()
+		if !cmp.Equal(tc.ttl, gotTTL, cmpopts.EquateApproxTime(time.Second)) {
```
Comment (hibiken) [inline client_test.go]: Use EquateApprox

File: client_test.go [hunk 5]
```diff
@@ -361,3 +363,210 @@ func TestClientEnqueueIn(t *testing.T) {
...
+			t.Errorf("TTL = %v, want %v", gotTTL, tc.ttl)
+			continue
+		}
+
+		// Enqueue the task again. It should fail.
+		err = c.Enqueue(tc.task, Unique(tc.ttl))
+		if err == nil {
+			t.Errorf("Enqueing %+v did not return an error", tc.task)
```
Comment (hibiken) [inline client_test.go]: typos here and below

**Pull request merged**